### PR TITLE
MXNet: Normalize rescale_grad in optimizer by Horovod size

### PR DESCRIPTION
**Pull request by yuxihu**
We currently perform average on gradients using allreduce methods in DistributedOptimizer and DistributedTrainer. This is equivalent to normalizing optimizer.rescale_grad by Horovod size(), which has better performance. 

This PR replaces allreduce average with normalized rescale_grad. Our tests show 2~3% performance boost when training ResNet50 with 8 GPUs.

### CODE REVIEW

**Comment by yuxihu** [comment]
@romerojosh @eric-haibin-lin @apeforest Please help review.

**Comment by yuxihu** [comment]
The test [failures](https://travis-ci.org/horovod/horovod/jobs/514525027) are not related to MXNet.

File: examples/mxnet_imagenet_resnet50.py [hunk 1]
```diff
@@ -317,6 +308,14 @@ def evaluate(epoch):
     if params is not None:
         hvd.broadcast_parameters(params, root_rank=0)
 
+    # Create optimizer
+    optimizer_params = {'wd': args.wd,
+                        'momentum': args.momentum,
+                        'lr_scheduler': lr_sched}
```
Comment (romerojosh) [inline examples/mxnet_imagenet_resnet50.py]: Does `rescale_grad` default to `1.0 / batch_size` for gluon but not module? Maybe it would be good to set it here anyway for consistency, or document the difference somehow.

Comment (yuxihu) [reply examples/mxnet_imagenet_resnet50.py]: This is a good question. Short answer is yes. For Gluon, we take care of it in [Trainer.step()](https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/gluon/trainer.py#L309) function so there is no need to set it in optimizer_params. But for Module, because we create optimizer first and wrap it with DistributedOptimizer, we need to specify [rescale_grad](https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/optimizer/optimizer.py#L51) in optimizer_params when creating optimizer.

I can add a comment here to document the difference.

**Comment by apeforest** [comment]
Can you also post performance numbers in this PR for future reference?

**Comment by alsrgv** [comment]
Love the removal of `rescale_grad` from the training script.  Restarted failed integration tests.

**Comment by yuxihu** [comment]
@alsrgv Looks like the tensorflow nightly [tests](https://travis-ci.org/horovod/horovod/jobs/514792270#L9894) failed.

**Comment by alsrgv** [comment]
We're fixing that separately, looks like TensorFlow changes the library name but did not update `tf.sysconfig.get_link_flags()`. 

@romerojosh @eric-haibin-lin @apeforest, can you stamp, so we can merge?
